unionml/remote.py

```diff
@@ -40,10 +41,23 @@ def create_project(remote: FlyteRemote, project: typing.Optional[str]):
         remote.client.register_project(Project(id=project, name=project, description=project))
 
 
-def get_app_version() -> str:
+def get_app_version(ignore_diff: bool = False) -> str:
     repo = git.Repo(".", search_parent_directories=True)
-    commit = repo.rev_parse("HEAD")
-    return commit.hexsha
+    if repo.is_dirty():
+        if not ignore_diff:
+            raise RuntimeError(
+                "You have uncommitted changes."
+                " Please commit your changes or explicitly ignore this using the --ignore-diff flag.",
+            )
+        logger.warning("You have uncommitted changes, unionml is operating based on the latest commit.")
+
+    with contextlib.suppress(git.CommandError):
+        if list(repo.iter_commits("@{upstream}..")):
+            logger.warning(
+                "You have local commits to not present on remote repositories, this may cause issues with deployment."
+            )
```

Comment:
This wasn't in the issue -- and it might not be necessary, but I don't know the ins-and-outs of Union / Flyte yet - but it seems to me that it might also be worthwhile to also check if there are local commits. 

In the dev-environment I set up for this, it's not really an issue since the build was happening locally. Is that always the case? Is it (or will it be possible) to build the container remotely?

Reply:
Suppression for the case when HEAD isn't part of a remote tracking branch.

Reply:
> but it seems to me that it might also be worthwhile to also check if there are local commits.

yeah I think a warning in this case is ok

> In the dev-environment I set up for this, it's not really an issue since the build was happening locally. Is that always the case? Is it (or will it be possible) to build the container remotely?

I'm not entirely sure... we may eventually support remote docker builds, but at this point we're gonna follow the Flyte convention where users build the image locally.

Reply:
> yeah I think a warning in this case is ok

Would you like this message to be changed?

Reply:
> You have local commits to not present on remote repositories, this may cause issues with deployment

Did you mean to say

> You have local commits that are not present on remote repositories, which may cause issues with deployment

Reply:
I did hahah, and even with prompting here I didn't notice the typo. I'll go with that ^